src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -848,15 +850,26 @@ export default class AircraftModel {
      * @return {boolean}
      */
     isBeyondTopOfDescent() {
-        const waypointModel = this.fms.nextHardAltitudeRestrictedWaypoint;
-
-        if (_isNil(waypointModel)) {
+        if (_isEmpty(this.fms.waypoints)) {
             return;
         }
 
-        const waypointAltitude = waypointModel.altitudeMaximum;
-        const waypointDistance = this.positionModel.distanceToPosition(waypointModel.positionModel);
-        const altitudeChange = waypointAltitude - this.altitude;
+        const currentAltitude = this.altitude;
+        const waypoints = this.fms.getAltitudeRestrictedWaypoints();
+        const waypointsWithRelevantCeiling = waypoints.filter((wp) => {
+            return (wp.altitudeMaximum !== INVALID_NUMBER) &&
+                (wp.altitudeMaximum < this.altitude)
+        }, this);
+
+        const targetAltitude = _isEmpty(waypointsWithRelevantCeiling) ?
+              this.fms.getBottomAltitude() :
+              waypointsWithRelevantCeiling[0].altitudeMaximum;
+        const targetPosition = _isEmpty(waypointsWithRelevantCeiling) ?
+              _last(this.fms.waypoints).positionModel :
+              waypointsWithRelevantCeiling[0].positionModel;
```

Comment:
If you're using `_last` to get the last item in an array you should also use `_head` to get the first item of an array.  Consistency.

https://lodash.com/docs/4.17.4#head